soh/soh/OTRGlobals.cpp

```diff
@@ -1470,14 +1605,62 @@ extern "C" CustomMessageEntry Randomizer_GetHintFromCheck(RandomizerCheck check)
     return hintText;
 }
 
+extern "C" GetItemEntry ItemTable_Retrieve(int16_t getItemID) {
+    if (OTRGlobals::Instance->getItemModIndex != MOD_NONE) {
+        getItemID++; // counteracts the - 1 offset used for vanilla table
+    }
+
+    return ItemTableManager::Instance->RetrieveItemEntry(OTRGlobals::Instance->getItemModIndex, getItemID);
+}
+
+extern "C" GetItemEntry ItemTable_RetrieveEntry(s16 tableID, s16 getItemID) {
+    if (tableID == MOD_NONE) {
+        getItemID--; // counteracts the - 1 offset used for vanilla table
+    }
+    return ItemTableManager::Instance->RetrieveItemEntry(tableID, getItemID);
+}
```

Comment:
it's not clear to me why we need to do the `--` when it's vanilla, and the `++` when it's not, mostly it's unclear how a `--` would counteract a `-1`

Reply:
Oh I think that comments needs to be changed. Vanilla GetItemIds have GI_NONE as the first entry, but the first sGetItemTable entry corresponds to the second GI value, and all the code that was pulling from sGetItemTable was calling getItemId - 1. But the rando table doesn't have that problem so it doesn't need the - 1. The one doing ++ adds it back when it's a rando item. The other function is actually doing the opposite, subtracting one when it's _not_ a rando item. I'll fix the comment, or after some more cleanup it's possible we don't to adjust that value at all anymore, even for vanilla items.

Reply:
Addressed by 69086f58c2b6bb00a273eefed4744ad456bba6ba